Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions.tsx

```diff
@@ -191,28 +195,55 @@ const SubmissionsProvider = ({ children }: ISubmissionsProviderProps) => {
         [ currentProblem, selectSubmissionTypeById, selectedSubmissionType ],
     );
 
+    const getProblemSubmissionError = useCallback(
+        (error: IErrorDataType) => {
+            const { detail, instance: problemId } = error;
+            const problemErrorId = parseInt(problemId, 10);
+            if (isNil(currentProblem) || problemErrorId !== currentProblem.id) {
+                return null;
+            }
+
+            return detail;
+        },
+        [ currentProblem ],
+    );
+
     useEffect(
         () => {
             if (!isNil(errorSubmitCode)) {
-                const { detail } = errorSubmitCode;
-                setSubmitMessage(detail);
-                return;
+                const error = getProblemSubmissionError(errorSubmitCode);
+                if (alertBoxSubmitMessage === CLOSED_ALERT_BOX_SUBMIT_MESSAGE) {
+                    setSubmitMessage(null);
+                } else {
+                    setSubmitMessage(error);
+                }
             }
 
             if (!isNil(errorSubmitFile)) {
-                const { detail } = errorSubmitFile;
-                setSubmitMessage(detail);
+                const error = getProblemSubmissionError(errorSubmitFile);
+                if (alertBoxSubmitMessage === CLOSED_ALERT_BOX_SUBMIT_MESSAGE) {
```

Comment:
Because if I go to another task after I close the error message and then back to the one with the error, the error message shows up again. This is because the error is still there if we do not make a new submit that updates the response with new data and the submit message is set to the error message each time we go back to the task.

Reply:
Then make the close button set a display:none style and clear the submt message state, so the message box disappears visibly and does not appear again because the state is clear.